tools/clang/lib/SPIRV/SPIRVEmitter.cpp

```diff
@@ -4475,7 +4475,19 @@ uint32_t SPIRVEmitter::castToInt(const uint32_t fromVal, QualType fromType,
   if (isBoolOrVecOfBoolType(fromType)) {
     const uint32_t one = getValueOne(toIntType);
     const uint32_t zero = getValueZero(toIntType);
-    return theBuilder.createSelect(intType, fromVal, one, zero);
+    if (toIntType->isScalarType() && toIntType->isLiteralType(astContext)) {
```

Comment:
Just a nit: this could be simpler to read: (change intType so it's not const)
```cpp
if (toIntType->isScalarType() && toIntType->isLiteralType(astContext))
  intType = toIntType->isSignedIntegerType() ? theBuilder.getInt32Type() : theBuilder.getUint32Type();
return theBuilder.createSelect(intType, fromVal, one, zero);
```

Reply:
Done